core/string/split_spec.rb

```diff
@@ -298,6 +298,8 @@
     "hi!".split(/()/, -1).should == ["h", "", "i", "", "!", "", ""]
     "hello".split(/((el))()/).should == ["h", "el", "el", "", "lo"]
     "AabB".split(/([a-z])+/).should == ["A", "b", "B"]
+    "aBBa".split(/(B)/, 2).should == ["a", "B", "Ba"]
+    "aBBa".split(/(B)/, 3).should == ["a", "B", "", "B", "a"]
```

Comment:
Could you move these examples to a separate example (`it`) to emphase how they deal with the limit?